salt/modules/http.py

```diff
@@ -114,3 +122,56 @@ def update_ca_bundle(target=None, source=None, merge_files=None):
     return salt.utils.http.update_ca_bundle(
         target, source, __opts__, merge_files
     )
+
+
+def download(target, source, hash=False, hash_type='sha256'):
+    '''
+    Download http file to the host file system.
+
+    target
+        Absolute path to target file.
+
+    source
+        The http source.
+
+    hash: False
+        If set to False, the hash of the downloaded file is not verified, else, the hash of the file.
+
+    hash_type: 'sha256'
+        The type of the hash.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' http.download /tmp/file.tar.gz http://host/file-123.tar.gz
+
+    '''
+    ret = {'Error': "Unable to load the 'urllib.request' library."}
+    if HAS_URLLIB_REQUEST:
+        if not salt.utils.path.is_absolute(target):
+            ret = {'Error': "target: '{}' have to be an absolute path.".format(target)}
+        elif salt.utils.path.is_dir(target):
+            ret = {'Error': "target: '{}' have to be a file path.".format(target)}
+        else:
+            if salt.utils.path.is_file(target):
```

Comment:
This would be more concise as an `elif`

Reply:
Lines 162 to 175 are common for the last condition.